sdk/cosmos/azure-cosmos/samples/session_token_management.py

```diff
@@ -0,0 +1,146 @@
+        target_feed_range = container.feed_range_from_partition_key(item['pk'])
+        response = container.create_item(item, session_token=previous_session_token)
+        session_token = response.get_response_headers()[HttpHeaders.SessionToken]
+        # adding everything in the cache in case consolidation is possible
+        for feed_range_json, session_token_cache in cache.items():
+            feed_range = json.loads(feed_range_json)
+            feed_ranges_and_session_tokens.append((feed_range, session_token_cache))
+        feed_ranges_and_session_tokens.append((target_feed_range, session_token))
+        latest_session_token = container.get_latest_session_token(feed_ranges_and_session_tokens, target_feed_range)
+        # only doing this for the key to be immutable
+        feed_range_json = json.dumps(target_feed_range)
+        cache[feed_range_json] = latest_session_token
+        previous_session_token = session_token
+
+
+def storing_session_tokens_container_feed_ranges(container):
+    print('2. Storing session tokens in a cache by feed range from the container.')
+
+    # The cache is a dictionary here for simplicity but in a real-world scenario, it would be some service.
+    cache = {}
+
+    # Everything below is just a simulation of what could be run on different machines and clients
+    # to store session tokens in a cache by feed range from the partition key.
+    feed_ranges_and_session_tokens = []
+    previous_session_token = ""
+    feed_ranges = list(container.read_feed_ranges())
+
+    # populating cache with session tokens
+    for i in range(5):
+        item = {
+            'id': 'item' + str(uuid.uuid4()),
+            'name': 'sample',
+            'pk': 'A' + str(random.randint(1, 10))
+        }
+        feed_range_from_pk = container.feed_range_from_partition_key(item['pk'])
+        response = container.create_item(item, session_token=previous_session_token)
+        session_token = response.get_response_headers()[HttpHeaders.SessionToken]
+        # adding everything in the cache in case consolidation is possible
+
+        for feed_range_json, session_token_cache in cache.items():
+            feed_range = json.loads(feed_range_json)
+            feed_ranges_and_session_tokens.append((feed_range, session_token_cache))
+        target_feed_range = next(
+            (feed_range for feed_range in feed_ranges if container.is_feed_range_subset(feed_range, feed_range_from_pk)),
+            {}
+        )
+        feed_ranges_and_session_tokens.append((target_feed_range, session_token))
+        latest_session_token = container.get_latest_session_token(feed_ranges_and_session_tokens, target_feed_range)
+        # only doing this for the key to be immutable
+        feed_range_json = json.dumps(target_feed_range)
+        cache[feed_range_json] = latest_session_token
+        previous_session_token = session_token
+
+
+def run_sample():
+    with CosmosClient(HOST, {'masterKey': MASTER_KEY}) as client:
```

Comment:
Can we make sure to only use `DefaultAzureCredential` in the samples going forward? The samples should demonstrate best practice.
(We should make a note to do this for existing samples too at some point).

Reply:
Changed